core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -538,23 +571,16 @@ class HttpGoogleServicesDAO(
     }
   }
 
-  /**
-    * NOTE: This function will returns "false" in both of the following cases:
-    * if you don't have sufficient scopes
-    *   - Google's JSON response body will contain "message": "Request had insufficient authentication scopes."
-    * if you're not authorized to see the billing account
-    *   - Google's JSON response body will contain "message" : "The caller does not have permission"
-    */
-  protected def credentialOwnsBillingAccount(credential: Credential, billingAccountName: String): Future[Boolean] = {
+  protected def billingAccountContainsFCBillingGroup(credential: Credential, billingAccountId: String): Future[Boolean] = {
```

Comment:
this function tests very specifically that there exists a policy with `billingGroupEmail` having `roles/billing.user`. It will not work if the email is granted the appropriate permissions in any other way: roles/billing.admin, a custom role, a more complicated group hierarchy.

Can we use https://cloud.google.com/billing/reference/rest/v1/billingAccounts/testIamPermissions instead to test that the service account we care about has appropriate permissions? This would mean pulling in a key for that SA because we need to auth as that identity.

Reply:
So the `projectnumber@cloudservices.gserviceaccount.com` SA isn't one we can get keys for in order to auth as. I think our options here are:

1. Test for one of `terra-billing@terra.bio` and `terra-billing@firecloud.org` being in the bindings, and increase their scope to "at least BA User or higher"
2. Put the Rawls SA in the group and see if _it_ has permissions (but that's back to the Rawls SA being mega-powerful again)
3. I don't know if Google has an endpoint similar to `testIamPermissions` but for a different SA. I couldn't find one.

Reply:
IMO we want to test whether the user has done what we ask of them. If they somehow found out the `projectnumber@cloudservices` address and added that, then DM would _work_ but wouldn't be resilient to any changes we'd make further down the line. I take your point about more complicated group hierarchies, though.

Reply:
I did a bit more research. A possible new option:

4. We might be able to generate a [short-lived service account credential](https://cloud.google.com/iam/docs/creating-short-lived-service-account-credentials) for the `projectnumber@cloudservices` such that Rawls can temporarily become that SA and test IAM permissions.

Reply:
I've bugged this as [CA-314](https://broadworkbench.atlassian.net/browse/CA-314). We're going to have to solve this before release because Lukas is giving terra-billing@ permissions at the Broad org-level, which doesn't show in the BA IAM policy and will thus incorrectly fail this check.